### PR TITLE
Move `Selector` into database crate

### PR DESCRIPTION
**Pull request by s7tya**
To implement #1734, it would be nice to move `Selector` from the site crate to the database crate.
This PR will move Selector itself and related things into the database crate.

I left things like SiteCtxt that is only used in the site crate as it is.

### CODE REVIEW

File: database/src/lib.rs [hunk 1]
```diff
@@ -245,7 +245,7 @@ impl fmt::Display for Profile {
 ///
 /// These are usually reported to users in a "flipped" way. For example,
 /// `Cache::Empty` means we're doing a "full" build. We present this to users as "full".
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, serde::Serialize)]
```
Comment (Kobzol) [inline database/src/lib.rs:252]: I think that the `serde::Serialize` part is also unused. You can remove it in another PR if you remember.

File: site/src/db.rs [hunk 2]
```diff
@@ -1,61 +1,2 @@
-use std::fmt;
-
 pub use crate::average::average;
```
Comment (Kobzol) [inline site/src/db.rs:1]: Let's delete the `db` file, it's useless now.

File: database/src/lib.rs [hunk 3]
```diff
@@ -9,6 +9,7 @@ use std::ops::{Add, Sub};
 use std::sync::Arc;
 use std::time::Duration;
 
+pub mod comparison;
```
Comment (Kobzol) [inline database/src/lib.rs:12]: Let's call it `metric`, since it only contains the metric enum right now :)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -109,14 +109,12 @@ jobs:
...
-        PROFILES: [
-          "Check,Debug,Doc",
-          "Opt",
-        ]
+        BENCH_INCLUDE_EXCLUDE_OPTS:
+          [
+            "--include cargo-0.60.0,stm32f4-0.14.0,webrender-2022",
+            "--exclude cargo-0.60.0,stm32f4-0.14.0,webrender-2022",
+          ]
+        PROFILES: ["Check,Debug,Doc", "Opt"]
```
Comment (Kobzol) [inline .github/workflows/ci.yml:117]: `["Check,Debug,Doc","Opt"]` :)

**Review by Kobzol — COMMENTED** [review]
Thanks! LGTM, apart from a few small nits. Would you like to also remove `async_trait` in this PR?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (s7tya) [reply .github/workflows/ci.yml]: My editor automatically formats them, so I just remove the diff! Thanks.

**Review by Kobzol — APPROVED** [review]
Looks good, thanks! This should help us use queries from collector nicely.

**Pull request merged**